classes/UpgradeSelfCheck.php

```diff
@@ -194,101 +195,156 @@ public function getWarnings(): array
-                );
+                return [
+                    'value' => $this->translator->trans('Your store\'s root directory isn\'t writable. Provide write access to the user running PHP with appropriate permission & ownership.'),
+                    'list' => [],
+                ];
 
             case self::ADMIN_UPGRADE_DIRECTORY_NOT_WRITABLE:
-                return $this->translator->trans(
-                    'The "/admin/autoupgrade" directory isn\'t writable. Provide write access to the user running PHP with appropriate permission & ownership.'
-                );
+                return [
+                    'value' => $this->translator->trans('The "/admin/autoupgrade" directory isn\'t writable. Provide write access to the user running PHP with appropriate permission & ownership.'),
+                    'list' => [],
+                ];
 
             case self::SAFE_MODE_ENABLED:
-                return $this->translator->trans('PHP\'s "Safe mode" needs to be disabled.');
+                return [
+                    'value' => $this->translator->trans('PHP\'s "Safe mode" needs to be disabled.'),
+                    'list' => [],
+                ];
 
             case self::F_OPEN_AND_CURL_DISABLED:
-                return $this->translator->trans(
-                    'Files can\'t be downloaded. Enable PHP\'s "allow_url_fopen" option or install PHP extension "cURL".'
-                );
-
+                return [
+                    'value' => $this->translator->trans('Files can\'t be downloaded. Enable PHP\'s "allow_url_fopen" option or install PHP extension "cURL".'),
+                    'list' => [],
+                ];
             case self::ZIP_DISABLED:
-                return $this->translator->trans('Missing PHP extension "zip".');
-
+                return [
+                    'value' => $this->translator->trans('Missing PHP extension "zip".'),
+                    'list' => [],
+                ];
             case self::MAINTENANCE_MODE_DISABLED:
-                return $this->translator->trans(
-                    'Maintenance mode needs to be enabled. Enable maintenance mode and add your maintenance IP.'
-                );
-
+                if ($isWebVersion) {
+                    $maintenanceLink = Context::getContext()->link->getAdminLink('AdminMaintenance');
+                    $params = [
+                        '[1]' => '<a href=' . $maintenanceLink . ' target="_blank">',
+                        '[/1]' => '</a>',
+                    ];
+                } else {
+                    $params = [
+                        '[1]' => '',
+                        '[/1]' => '',
+                    ];
+                }
+
+                return [
+                    'value' => $this->translator->trans('Maintenance mode needs to be enabled. Enable maintenance mode and add your maintenance IP in [1]Shop parameters > General > Maintenance[/1].', $params),
+                    'list' => [],
```

Comment:
why do you add empty list ? I think we can just add list if needed no ?

Reply:
This saves me from having to check if there is a list in the command or in the template.

Reply:
Afterwards, it could be cleaner. I'm going to make this optional